lib/gollum-lib/filter/tags.rb

```diff
@@ -225,7 +225,7 @@ def process_page_link_tag(link_part, pretty_name = nil)
     end
     presence = :page_present if page
 
-    name = pretty_name ? pretty_name : link
+    name = pretty_name ? pretty_name : path_to_link_text(link, !!page)
```

Comment:
I added the second argument to `path_to_link_text` because I couldn't see a good way of writing it here. The logic needs to be this:

```ruby
if page
  name =  name = pretty_name ? pretty_name : path_to_link_text(link)
  link = page.escaped_url_path
else
  name =  name = pretty_name ? pretty_name : link
  link = ERB::Util.url_encode(link).force_encoding('utf-8')
end
```

But as you can see that involves some annoying duplication